### ISSUE TITLE
[BUG] Lazy reader error in passthrough benchmark: reference not allowed

### ISSUE THREAD

**Issue by gunnsth** [unidoc/unipdf#108]
## Description
When running pdf_passthrough_bench (https://github.com/unidoc/unipdf-examples/blob/v3/testing/pdf_passthrough_bench.go) on certain PDFs
with the lazy reader (instead of regular reader) there is an error message:

```
000280.pdf - fail reference not allowed
000671.pdf - fail reference not allowed
000839.pdf - fail reference not allowed
...
```

## Expected Behavior
Expect files to be processed the same with lazy reader as regular reader.  The regular reader processes the files successfully.

## Actual Behavior
1. Change `NewPdfReader` to `NewPdfReaderLazy`
2. Run the benchmark on the files, should see the error: reference not allowed

## Attachments
1. Code: https://github.com/unidoc/unipdf-examples/blob/v3/testing/pdf_passthrough_bench.go
2. Problematic files:
[reference-not-allowed.zip](https://github.com/unidoc/unipdf/files/3362430/reference-not-allowed.zip)

**Comment by gunnsth** [comment]
Ran into a few more cases where this happens. See attached:
[reference-not-allowed.zip](https://github.com/unidoc/unipdf/files/3414498/reference-not-allowed.zip)

**Comment by gunnsth** [comment]
Resolved in #131
